core/src/core/temporal/utils/execution.py

```
@@ -0,0 +1,161 @@
+        try:
+            result = await task
+            logger.info(
+                "Stage %s completed: %s records",
+                stage_name,
+                result.records_processed,
+            )
+            results.append(result)
+
+        except ActivityError as e:
+            failed_result = _create_activity_error_result(stage_name, e)
+            logger.error("Stage %s failed: %s", stage_name, str(e))
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)